### PR TITLE
Adding support for multiple databases

### LINKED ISSUE #46: Multiple database support for C5 implementation
**Description:**
Current C5 implementation of RDBMS based carbon-coordination only works with mysql. Need to add support for oracle, mssql and postgresql databases

**Suggested Labels:**
Feature Enhancement
Priority High

**Suggested Assignees:**
@anoukh

**Affected Product Version:**
carbon-coordination C5 implementation

**OS, DB, other environment details and versions:**    
N/A

**Steps to reproduce:**
N/A

**Related Issues:**
N/A

### PR DESCRIPTION
**Pull request by Anoukh**
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/pom.xml [hunk 1]
```diff
@@ -88,6 +92,7 @@
                             org.wso2.carbon.cluster.coordinator.commons.*,
                             com.google.common.*;version="${guava.imp.package.version}",
                             org.apache.commons.logging.*;version="${commons.logging.imp.package.version}",
+                            org.wso2.carbon.database.query.manager.*; version="${carbon.analytics-common.version.range}",
                             *;resolution:=optional
```
Comment (grainier) [inline components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/pom.xml]: can we get rid of `*;resolution:=optional`?

File: components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/pom.xml [hunk 2]
```diff
@@ -88,6 +92,7 @@
...
                         </Import-Package>
                         <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (grainier) [inline components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/pom.xml]: try to get rid of `DynamicImport`

**Review by mohanvive — APPROVED** [review]

File: components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java [hunk 3]
```diff
@@ -98,97 +110,88 @@ public RDBMSCommunicationBusContextImpl(DataSource dataSource) {
...
-            connection.commit();
-            if (log.isDebugEnabled()) {
-                log.debug("Membership Event Table Created Successfully");
+            try {
+                con = getConnection();
+                con.setAutoCommit(false);
+                stmt = con.createStatement();
+            } catch (SQLException e) {
+                log.error("Cannot establish connection to the database to create " + tableName + " if not exists", e);
+                return;
```
Comment (nirmal070125) [inline components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java:149]: Why don't we close the connection? Why are we swallowing the exception?

Comment (Anoukh) [reply components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java:149]: This is wrapped in a try finally block, where the connection will be closed.

Comment (Anoukh) [reply components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCommunicationBusContextImpl.java:149]: Fixed to throw exception in commit cf4cdfa

**Pull request merged**